### PR TITLE
use Jenkins test harness for testing

### PR DESCRIPTION
**Pull request by daspilker**
This is work in progress and need changes from https://github.com/jenkinsci/job-dsl-plugin/pull/837.

### CODE REVIEW

**Comment by sheehan** [comment]
@daspilker is this ready for merge?

**Comment by daspilker** [comment]
Yes, it's ready.

File: build.gradle [hunk 1]
```diff
@@ -14,14 +14,34 @@ repositories {
...
+    testCompile "org.jenkins-ci.main:jenkins-war:${jenkinsVersion}:war-for-test@jar"
+
+    // Job DSL plugin including plugin dependencies
+    testCompile "org.jenkins-ci.plugins:job-dsl:${jobDslVersion}"
+    testCompile "org.jenkins-ci.plugins:job-dsl:${jobDslVersion}@jar"
+    testCompile 'org.jenkins-ci.plugins:structs:1.1@jar'
+    testCompile 'org.jenkins-ci.plugins:cloudbees-folder:4.4@jar'
+
+    // plugins to install in test instance
+    testPlugins 'org.jenkins-ci.plugins:ghprb:1.31.4'
```
Comment (Tin-Nguyen) [inline build.gradle]: can this resolve the issue with Generated DSL method? I tried to follow the configuration here, but still got the issue
```
javaposse.jobdsl.dsl.DslScriptException: (PullRequestJobTemplate.groovy, line 59) No signature of method: javaposse.jobdsl.dsl.helpers.triggers.TriggerContext.githubPullRequest() is applicable for argument types: (com.unified.dsl.templates.PullRequestJobTemplate$_closure1$_closure5$_closure15) values: [com.unified.dsl.templates.PullRequestJobTemplate$_closure1$_closure5$_closure15@40883107]
```

Comment (Tin-Nguyen) [reply build.gradle]: the method `githubPullRequest` is provided by `ghprb` plugin, I tried to at the plugin into `testPlugins`, but the unit test still got failure

**Pull request merged**